sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionsAsyncClient.java

```diff
@@ -0,0 +1,335 @@
+     *                         type: String (Required)
+     *                         function (Required): {
+     *                             name: String (Required)
+     *                             arguments: String (Required)
+     *                         }
+     *                     }
+     *                 ]
+     *             }
+     *         }
+     *     ]
+     * }
+     * }</pre>
+     *
+     * @param completeRequest The completeRequest parameter.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return chat completions for the provided chat messages.
+     * Completions support a wide variety of tasks and generate text that continues from or "completes"
+     * provided prompt data along with {@link Response} on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    private Mono<Response<BinaryData>> completeWithBinaryResponse(BinaryData completeRequest,
+        RequestOptions requestOptions) {
+        return this.serviceClient.completeWithResponseAsync(completeRequest, requestOptions);
+    }
+
+    /**
+     * Gets chat completions for the provided chat messages.
+     * Completions support a wide variety of tasks and generate text that continues from or "completes"
+     * provided prompt data. The method makes a REST API call to the `/chat/completions` route
+     * on the given endpoint.
+     *
+     * @param options The configuration information for a chat completions request. Completions support a
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return chat completions for the provided chat messages.
+     * Completions support a wide variety of tasks and generate text that continues from or "completes"
+     * provided prompt data along with {@link Response} on successful completion of {@link Mono}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<ChatCompletions>> completeWithResponse(ChatCompletionsOptions options) {
```

Comment:
We can drop this method as we already have `Mono<ChatCompletions> complete(ChatCompletionOptions options)` method below. We can add this "WithResponse" method later, if there's a customer request to support this.

Reply:
[Updated](https://github.com/Azure/azure-sdk-for-java/pull/41482/commits/e16676f12ecb82d477715c3343df29291a46e831)